src/main/java/duke/parser/DataParser.java

```diff
@@ -0,0 +1,258 @@
+package duke.parser;
+
+import duke.command.Command;
+import duke.command.AddDeadlineTaskCommand;
+import duke.command.AddEventTaskCommand;
+import duke.command.AddToDoTaskCommand;
+import duke.command.CompleteTaskCommand;
+import duke.command.DeleteTaskCommand;
+import duke.command.EndCommand;
+import duke.command.ListTaskCommand;
+import duke.command.FindTaskCommand;
+
+import java.util.Scanner;
+import duke.exception.DukeException;
+import duke.exception.UnknownCommandException;
+import duke.exception.InvalidTaskIndexException;
+import duke.exception.InvalidDeadlineException;
+import duke.exception.InvalidEventException;
+import duke.exception.InvalidToDoException;
+import duke.exception.InvalidKeywordException;
+
+/**
+ * Represents a Data Parser to parse in all user input provided.
+ */
+public class DataParser {
+
+    private Scanner sc;
+    private String input;
+
+    /**
+     * Constructs a new DataParser to read in the user input.
+     */
+    public DataParser() {
+        this.sc = new Scanner(System.in);
+        this.input = "";
+    }
+
+    /**
+     * To check the user has anymore data left unread.
+     * If there is anymore data left unread, return true.
+     * @return Returns true if there is data left unread.
+     */
+    public boolean hasAnymoreData() {
+        return sc.hasNextLine();
+    }
+
+    /**
+     * Reads the input line.
+     */
+    public void readInput() {
+        this.input = sc.nextLine();
+    }
+
+    /**
+     * Reads the given input line.
+     * @param input The given input line.
+     */
+    public void readInput(String input) {
+        this.input = input;
+    }
+
+    /**
+     * Parses the User Input data and returns a Command based on the first word given by the user.
+     * Inputs should start with "bye", "delete", "done" etc.
+     * @return A Command based on the user input.
+     * @throws DukeException If the user input is invalid.
+     */
+    public Command findCommand() throws DukeException {
```

Comment:
Maybe you could rename this method because there is a FindTaskCommand as well and it was just a little confusing

Reply:
Hi Yi Yin! 

Will rename in the next commit, thanks!